### PR TITLE
[entt] Update to 3.4.0 (#11507)

### LINKED ISSUE #11507: [entt] update to 3.4.0
Library name: entt

New version number: 3.4.0

Other information that may be useful: https://github.com/skypjack/entt/releases/tag/v3.4.0

### PR DESCRIPTION
**Pull request by Vennor**
Fixes #11507

### CODE REVIEW

File: ports/entt/portfile.cmake [hunk 1]
```diff
@@ -19,7 +19,7 @@ vcpkg_install_cmake()
 if(EXISTS ${CURRENT_PACKAGES_DIR}/cmake)
     vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
 else()
-    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/entt)
+    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/EnTT/cmake)
 endif()
```
Comment (NancyLi1013) [inline ports/entt/portfile.cmake:24]: Could you please update the way to handle copyright as 
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/entt/portfile.cmake (hunk 1 above)
Comment (Vennor) [reply ports/entt/portfile.cmake:24]: Sure! Done.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@Vennor thanks for this PR.
LGTM now.

**Comment by strega-nil** [comment]
Thank you @Vennor!

**Pull request merged**